osu.Game/Screens/Edit/Compose/Components/Timeline/Timeline.cs

```diff
@@ -82,15 +82,16 @@ public Timeline(Drawable userContent)
         private Container mainContent;
 
         private Bindable<float> waveformOpacity;
-        private Bindable<bool> controlPointsVisible;
+        private Bindable<bool> alwaysShowControlPoints;
+        private readonly Bindable<bool> controlPointsVisible = new Bindable<bool>();
```

Comment:
I don't think you need this new bindable, and can revert basically all your changes.

In the original `controlPointsVisible.BindValueChanged` you can just check the current editor mode and react there. Although one may argue this is too much local knowledge and the *editor* should be setting the value on the *timeline*, not vice-versa.

Reply:
i'm unclear on how this would work, do you mind explaining to me why this is the case?
suppose we have the case of:
```
actions    | 1. start with config hidden -> 2. enable config -> 3. change to timing mode -> 4. disable config -> 5. go to compose mode
visibility | 1. hidden -> 2. shown -> 3. shown -> 4. shown -> 5. hidden
height     | 1. normal -> 2. expanded -> 3. expanded -> 4. normal -> 5. normal
```
if we don't create a bindable on mode state, at what point can code trigger at action (5) to correctly hide the config? is there a lifecycle thing here i'm missing..?

Reply:
Check https://github.com/ppy/osu/pull/29211, I've redone this feature in a slightly simpler way.